cachito/web/models.py

```diff
@@ -1,11 +1,69 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
+from copy import deepcopy
 
 from cachito.web import db
 
 
+request_pkg_manager_table = db.Table(
+    'request_pkg_manager',
+    db.Model.metadata,
+    db.Column('request_id', db.Integer, db.ForeignKey('request.id')),
+    db.Column('pkg_manager_id', db.Integer, db.ForeignKey('pkg_manager.id'))
+)
+
+
 class Request(db.Model):
     """A Cachito user request."""
+    __tablename__ = 'request'
     id = db.Column(db.Integer, primary_key=True)
+    git_repo = db.Column(db.String, nullable=False)
+    git_ref = db.Column(db.String, nullable=False)
+    pkg_managers = db.relationship('PackageManager', secondary=request_pkg_manager_table)
 
     def __repr__(self):
         return '<Request {0!r}>'.format(self.id)
+
+    def to_json(self):
+        pkg_managers = [pkg_manager.to_json() for pkg_manager in self.pkg_managers]
+        return {
+            'id': self.id,
+            'git_repo': self.git_repo,
+            'git_ref': self.git_ref,
+            'pkg_managers': pkg_managers,
+        }
+
+    @classmethod
+    def from_json(klass, kwargs):
+        # Validate all required parameters are present
+        required_params = {'git_repo', 'git_ref', 'pkg_managers'}
+        missing_params = required_params - set(kwargs.keys())
+        if missing_params:
+            raise ValueError('Missing required parameter(s): {}'.format(', '.join(missing_params)))
+        kwargs = deepcopy(kwargs)
+
+        # Validate package managers are correctly provided
+        pkg_managers = kwargs.pop('pkg_managers', None)
+        if not pkg_managers:
+            raise ValueError('At least one package manager is required')
+        for name in set(pkg_managers):
+            pkg_manager = PackageManager.query.filter_by(name=name).first()
```

Comment:
You might be able to replace this for loop with a single query using `_in` and then compare the length of `set(pkg_managers)` and the results.

Reply:
Done! (Although, I don't think the code is much cleaner with that approach)